core/src/main/scala/scala/pickling/generator/sourcegen.scala

```diff
@@ -0,0 +1,577 @@
+package scala.pickling
+package generator
+
+import scala.reflect.api.Universe
+
+/** This is the portion of code which can actually generate a pickler/unpickler object instance
+  * from the IR AST.
+  *
+  *
+  */
+private[pickling]  trait SourceGenerator extends Macro with FastTypeTagMacros {
+  import c.universe._
+
+  def pickleNull(builder: c.Tree): c.Tree =
+    q"""$builder.hintTag(_root_.scala.pickling.FastTypeTag.Null)
+        _root_.scala.pickling.Defaults.nullPickler.pickle(null, $builder)"""
+
+
+
+  def allowNonExistentField(impl: Tree): c.Tree = {
+    q"""try $impl catch {
+            case _: _root_.scala.pickling.PicklingException =>
+            // TODO - Don't just ignore bad things, figure out how to actually read the class file for reals.
+            // We could basically by blocked by scala here, though.
+          }"""
+  }
+  def generatePickleImplFromAst(picklerAst: PicklerAst): c.Tree = {
+
+    def genGetField(x: GetField): c.Tree = {
+      def pickleLogic(isStaticallyElided: Boolean, fieldValue: Tree): Tree = {
+        val elideHint =
+          if (isStaticallyElided) q"b.hintStaticallyElidedType()" else q""
+        // TODO - Hint dynamically elided?
+        // NOTE; This will look up an IMPLICIT pickler for the value, based on the type.
+        //       This is how we chain our macros to find all the valid types.
+
+        q"""
+           $elideHint
+           _root_.scala.pickling.functions.pickleInto($fieldValue, b)"""
+      }
+
+      def putField(getterLogic: Tree, isStaticallyElided: Boolean): c.Tree =
+        q"builder.putField(${x.name}, b => ${pickleLogic(isStaticallyElided, getterLogic)})"
+
+      x.getter match {
+        case x: IrField =>
+          // TODO - handle
+          val tpe = x.tpe[c.universe.type](c.universe)
+          val staticallyElided = tpe.isEffectivelyFinal || tpe.isEffectivelyPrimitive
+          if(x.isScala || !x.isPublic) {
```

Comment:
Is this code reachable if the user selected "static only" algorithms? If so could we fail the compilation at this point if there's static only flag?